tools/project_manager/external_catalog_search.php

```diff
@@ -129,186 +217,227 @@ function do_search_and_show_hits()
-            ['label' => _("LCCN"),      'value' => $marc_record->lccn],
-            ['label' => _("ISBN"),      'value' => $marc_record->isbn],
-        ] as $couple
-        ) {
-            $label = $couple['label'];
-            $value = $couple['value'];
-            echo "<tr>";
-            echo   "<th class='left-align top-align' style='width: 20%;'>$label:</th>";
-            echo   "<td class='left-align top-align'>$value</td>";
-            echo "</tr>\n";
+    // Display results
+    echo "<p>", sprintf(_("%d results returned."), $total_hits), "</p>\n";
+
+    if ($total_hits > 0) {
+        $encoded_fullquery = base64_encode(serialize($fullquery));
+        $url_base = "external_catalog_search.php?action=do_search_and_show_hits&fq=$encoded_fullquery&" . search_query_params();
+        if ($hide_nontext) {
+            $url_base .= "&hide_nontext=on";
         }
 
-        echo "</table><p></td>";
+        // PHPStan up to at least 1.10.57 has a bug where it doesn't correctly detect
+        // that $num_nontext is updated in the loop. We need to sprinkle several ignores
+        // to get it to ignore this.
+        // @phpstan-ignore-next-line
+        $book_frag = ($hide_nontext && $num_nontext > 0) ? sprintf(_(" (%d non-textual hidden)"), $num_nontext) : "";
+
+        if ($num_nontext < $hits_per_page) { /** @phpstan-ignore-line */
+            echo "<p>", _("Please pick a result from below:"), "</p>\n";
+
+            display_navbar($url_base, $start, $hits_per_page, $total_hits, $book_frag);
+
+            // Display the results as a two column, multi-row table where each cell is
+            // an input form radio button with a single subtable of label/value rows.
+            echo "<form method='post' action='editproject.php'>\n";
+            echo "<input type='hidden' name='action' value='create_from_marc_record'>\n";
+            echo "<table style='width: 100%; border: 0;'>\n";
+
+            $i = 1;
+            foreach ($marc_records as [$r, $m]) { /** @phpstan-ignore-line */
+                if ($i % 2 == 1) {
+                    echo "<tr>";
+                }
+                // Radio button to select record
+                echo "<td class='center-align top-align' style='width: 5%;'>";
+                echo "<input style='margin-top:1.6em' type='radio' name='rec' value='".base64_encode(serialize($r))."'>";
+                echo "</td>";
+
+                // Subtable for record
+                echo "<td class='left-align top-align' style='width: 45%;'>";
+
+                // Compresss the record so we don't run over URL length limits
+                $rec = base64url_encode(gzencode(serialize($r), 9));
+                echo "<div style='text-align:right; font-size:small'>\n";
+                echo   "<a style='text-decoration: none' target='_blank' href='marc_inspector.php?rec=$rec'>\n";
+                echo     "inspect MARC <i class='fa fa-external-link-alt'></i>\n";
```

Comment:
re: FA icon - I like this. We should replace our custom "open in a new window" icon with this site-wide. Not in this PR, obvs. Probably should think about doing this for all or our little inline graphics.

Reply:
Thanks. Ray pointed me in the right direction. The minor niggle I have about this is I don't like the size. I couldn't figure out how to make it size proportional to the `font-size` of the `div`.